components-api/src/main/scala/pl/touk/nussknacker/engine/api/component/SingleComponentConfig.scala

```diff
@@ -69,26 +69,26 @@ object ParameterConfig {
   val empty: ParameterConfig = ParameterConfig(None, None, None, None, None)
 }
 
-@JsonCodec case class ScenarioPropertyConfig(
+@JsonCodec case class SingleScenarioPropertyConfig(
```

Comment:
I'm not a fan of naming thing with the `Single` prefix. It breaks the encapsulation / interface segregation principle. The `scenario property config` name already saying that there is only one property. You added `Single` to resolve problem in the place where this class is used so you have cycle of dependencies: name of class depends on the place of usage which depends on the class.   

Also, please move this class to `deployment-manager-api`. This class isn't used by components, it is used by deployment managers which can defined properties config programmatically in `DeploymentManagerProvider.scenarioPropertiesConfig`

Reply:
It makes it easier to make the distinciton between a single property and properties which are pretty close ot each other in terms of naming. I don't agree that the name depends on the place - even if we removed the Single prefix, it would still be refering to a singular entity. It is also in line with the `SingleComponentConfig`.

As for moving the class, I tried moving it to `nussknacker-deployment-manager-api` but there are a few classes in `components-api` that use this class and they break after this change. Should i somehow reroute the packages so that the classes from `deployment-manager-api` can be used in components or do we leave this?

Reply:
I have renamed this and also the SingleComponentConfig class.